beer-garden/beer-garden
# #1091 - Added sync_roles function for importing roles

Closes #1091 

This PR adds the following:

* An example config, `roles.yaml`, which will eventually act as a reasonable set of default roles.
* A configuration parameter under `auth` called `role_definition_file` that you set to the location of your `roles.yaml` equivalent file.
* A `sync_roles` function that takes a dictionary containing role info and syncs the Roles in the database to match those definitions.
* A modified version of `ensure_roles` that runs on application startup.  This reads the yaml file referred to in the `auth.role_definition_file` configuration parameter and passes it over to the `sync_roles` function.

# Test Instructions
* Set the `auth.role_definition_file` configuration parameter in your application config file to the location of the `roles.yaml` example config file (or a copy of it if you house your config files elsewhere).
* Start beer-garden.
* Check that the Roles got created. You can do this in the DB directly, or by doing something like this in a python shell:
  ```python
  In [1]: from mongoengine import connect

  In [2]: from beer_garden.db.mongo.models import Role

  In [3]: connect(db="bg-parent1", host="mongodb://mongodb")  # change parameter values as appropriate for your setup
  Out[3]: MongoClient(host=['mongodb:27017'], document_class=dict, tz_aware=False, connect=True, read_preference=Primary())

  In [4]: [role.name for role in Role.objects.all()]
  Out[4]: ['job_manager', 'operator', 'read_only', 'superuser']
  ```
* Feel free to play around with the `roles.yaml` and verify that roles get added, removed, and modified when you expect them to.  Or just look at the unit tests and convince yourself that that would be the case.
